pkg/virtctl/usbredir/usbredir.go

```diff
@@ -0,0 +1,201 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017, 2020 Red Hat, Inc.
+ *
+ */
+
+package usbredir
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"os/signal"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	USBREDIRSERVER = "usbredirect"
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "usbredir (vendor:product) (resource name) (VMI)",
+		Short:   "Redirect a usb device to a virtual machine instance.",
+		Example: usage(),
+		Args:    templates.ExactArgs("usb", 3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := usbredirCommand{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type usbredirCommand struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func (usbredirCmd *usbredirCommand) Run(command *cobra.Command, args []string) error {
+	if _, err := exec.LookPath(USBREDIRSERVER); err != nil {
+		return fmt.Errorf("Error on finding %s in $PATH: %s", USBREDIRSERVER, err.Error())
+	}
+
+	namespace, _, err := usbredirCmd.clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	virtCli, err := kubecli.GetKubevirtClientFromClientConfig(usbredirCmd.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	vmiArg := args[2]
+	usbNameArg := args[1]
+	usbdeviceArg := args[0]
+
+	glog.V(2).Infof("FIXME: Handle USB label name properly: %v", usbNameArg)
```

Comment:
It looks weird to put a FIXME into a log message. Was the intent to fix this now or later?

Reply:
Thanks, that was a leftover. The usb label is being handled properly, that is, if you define label named `storage` for a -usb device in the yaml file, it creates `/run/kubevirt-private/$hash/virt-usbredir-storage` and that is what is used based on user command line argument, the $name when you run `virtctl usbredir $device $name $vmi`